File: 백준/Silver/15651. N과 M （3）/N과 M （3）.cc
#include <iostream>

using namespace std;

int arr[8];

int N, M;

void print(int m) {
	if (m == M) {
		for (int j = 0; j < M; j++) {
			cout << arr[j] << " ";
		}
		cout << "\n";
	}
	else {
		for (int i = 1; i <= N; i++) {
			arr[m] = i;
			print(m + 1);
			arr[m] = 0;
		}

	}
}

int main() {
	cin >> N >> M;

	print(0);
}

File: 백준/Silver/15651. N과 M （3）/N과 M （3）.py
import sys


def input(): return sys.stdin.readline().rstrip()


N, M = map(int, input().split())


def solution(arr, start, depth):
    if depth == M:
        print(*arr)
        return

    for i in range(start, N + 1):
        arr.append(i)
        solution(arr, start, depth + 1)
        arr.pop()


solution([], 1, 0)